slider/curve.py

```diff
@@ -128,13 +123,42 @@ def __call__(self, t):
         return self._curves[bi]((t - pre_t) / (post_t - pre_t))
 
 
-class Passthrough(Curve):
-    kinds = 'P'
+class LinearMetaCurve(MetaCurve):
+    kinds = 'L'
 
     def __init__(self, points, req_length):
         self.points = points
         self.req_length = req_length
-        self._center = center = get_center(*self.points)
+        self._curves = [
+            Bezier(subpoints, None) for subpoints in sliding_window(2, points)
+        ]
+
+
+class Perfect(Curve):
+    kinds = 'P'
+
+    def __new__(cls, points, req_length):
+        if len(points) != 3:
```

Comment:
It seems a bit weird to be returning a bezier instance when you called the perfect constructor. Wouldn't it be better to put this logic in the `from_kind_and_points` method? Then the user could forcibly get a perfect curve if they called the perfect constructor and would get a valueerror of the points were bad.